src/haddock/libs/libhpc.py

```diff
@@ -0,0 +1,214 @@
+"""Module in charge of running tasks in HPC."""
+import os
+import re
+import shlex
+import subprocess
+import time
+from pathlib import Path
+
+from haddock import log
+
+
+STATE_REGEX = r"JobState=(\w*)"
+
+
+class HPCWorker:
+    """Defines the HPC Job."""
+
+    def __init__(self, tasks, num):
+        self.tasks = tasks
+        log.debug(f"HPCWorker ready with {len(self.tasks)}")
+        self.job_id = None
+        self.job_num = num
+        self.job_status = "unknown"
+        self.job_fname = ""
+
+    def run(self):
+        """Execute the tasks."""
+        job_file = self.prepare_job_file(self.tasks)
+        cmd = f"sbatch {job_file}"
+        p = subprocess.run(shlex.split(cmd), capture_output=True)
+        self.job_id = int(p.stdout.decode("utf-8").split()[-1])
+        self.job_status = "submitted"
+
+    def update_status(self):
+        """Retrieve the status of this worker."""
+        cmd = f"scontrol show jobid -dd {self.job_id}"
+        p = subprocess.run(shlex.split(cmd), capture_output=True)
+        out = p.stdout.decode("utf-8")
+        # err = p.stderr.decode('utf-8')
+        if out:
+            # TODO: Maybe a regex here is overkill
+            # https://regex101.com/r/M2vbAc/1
+            status = re.findall(STATE_REGEX, out)[0]
+            if status == "PENDING":
+                self.job_status = "submitted"
+            elif status == "RUNNING":
+                self.job_status = "running"
+            elif status == "SUSPENDED":
+                self.job_status = "hold"
+            elif status == "COMPLETING":
+                self.job_status = "running"
+            elif status == "COMPLETED":
+                self.job_status = "finished"
+            elif status == "FAILED":
```

Comment:
Python :coffee: talk,
in these cases I prefer:

```python
job_status = {
    "PENDING": "submitted",
    "RUNNING": "running",
    "SUSPENDED": "hold",
    "COMPLETING": "running",
    "COMPLETED": "finished",
    "FAILED": "failed",
    }

self.job_status = job_status[status]
```

In this way, if you ever get a key you didn't code for, you get an error also. Otherwise with the `elif` it will pass silently.

Reply:
oh good suggestion!